currency-rates/internal/clients/chain.go

```diff
@@ -0,0 +1,40 @@
+package clients
+
+import (
+	"context"
+)
+
+type Chain interface {
+    RateClient
+    SetNext(Chain)
+}
+
+type BaseChain struct {
+    rateClient RateClient
+    next Chain
+}
+
+func NewBaseChain(rateClient RateClient) *BaseChain {
+    return &BaseChain{
+        rateClient: rateClient,
+    }
+}
+
+func (chain *BaseChain) SetNext(next Chain) {
+    chain.next = next
+}
+
+func (chain *BaseChain) GetDollarRate(ctx context.Context) (float64, error) {
+	rate, err := chain.rateClient.GetDollarRate(ctx)
+
+	if err != nil {
+		next := chain.next
```

Comment:
just use 
if chain.next == nil ....
no need to use separate variable

Reply:
done